app/src/main/java/com/hamdan/forzenbook/domain/usecase/login/LoginGetTokenFromNetworkUseCaseImpl.kt

```diff
@@ -0,0 +1,17 @@
+package com.hamdan.forzenbook.domain.usecase.login
+
+import android.util.Log
+import com.hamdan.forzenbook.data.repository.LoginRepository
+
+class LoginGetTokenFromNetworkUseCaseImpl(
```

Comment:
I don't like the naming here, similar to the repo methods naming. It doesn't really describe what's going on. This is a "LoginUseCase", the token is an implementation detail that doesn't need to be part of naming. the Database use case is "GetStoredCredentials", again with the token not needing to be part of the naming. 

Implementation details can be hidden away

Reply:
Should I then put all the Login-based Use cases under one Use case? I thought that was against the point of the Use cases. giving explicitly what's needed. The naming definitely feels wrong but how do I differentiate them then?

Reply:
ok actually I kind of see what your saying on re-reading the comment